pkg/agent/route/route_windows.go

```diff
@@ -281,27 +280,32 @@ func (c *Client) addServiceCIDRRoute(serviceCIDR *net.IPNet) error {
 
 	// Collect stale routes.
 	var staleRoutes []*util.Route
-	// If current destination CIDR is not nil, the route with current destination CIDR should be uninstalled since
-	// a new route with a newly calculated destination CIDR has been installed.
-	if serviceCIDRRouteExists {
-		staleRoutes = append(staleRoutes, oldServiceCIDRRoute.(*util.Route))
-	}
 	routes, err := c.listIPRoutesOnGW()
 	if err != nil {
 		return fmt.Errorf("error listing ip routes: %w", err)
 	}
-	// Collect stale per-IP routes for ClusterIPs before this patch.
 	for _, rt := range routes {
-		ones, _ := rt.DestinationSubnet.Mask.Size()
-		if ones == net.IPv4len*8 && serviceCIDR.Contains(rt.DestinationSubnet.IP) {
+		if !rt.GatewayAddress.Equal(gw) {
```

Comment:
Like linux implementation, I think the cleanup only needs to be executed one time?

Reply:
Done.